food-vendor/src/main/java/com/example/foodvendor/data/FoodVendorDAO.java

```diff
@@ -15,15 +20,29 @@
 
     private Map<Integer, Map<String, Ingredient>> vendorToIngredientsMap;
 
+    private static final Tracer tracer = Tracing.getTracer();
+
     public FoodVendorDAO() {
         setupVendorToIngredientsMap();
     }
 
     public Ingredient getIngredient(int id, String ingredient) {
+        Span span = tracer.spanBuilder("FoodVendor starting query...")
+                .setRecordEvents(true)
+                .setSampler(Samplers.alwaysSample())
+                .startSpan();
+
         Map<String, Ingredient> vendorInventory = vendorToIngredientsMap.get(id);
         if (vendorInventory == null) {
             return null;
```

Comment:
Does it matter that we return here without calling span.end()?

Reply:
Good catch, added `span.end()`